backend/Tasque.BLL/Services/Auth/AuthService.cs

```diff
@@ -0,0 +1,95 @@
+﻿using AutoMapper;
+using FluentValidation;
+using Microsoft.EntityFrameworkCore;
+using Tasque.Core.BLL.Exeptions;
+using Tasque.Core.BLL.JWT;
+using Tasque.Core.Common.DTO;
+using Tasque.Core.Common.Entities;
+using Tasque.Core.Common.Security;
+using Tasque.Core.DAL;
+
+namespace Tasque.Core.BLL.Services.Auth
+{
+    public class AuthService
+    {
+        private DataContext _context;
+        private JwtFactory _jwtFactory;        
+        private IMapper _mapper;
+        private IValidator<User> _validator;
+        private ConfirmationTokenService _tokenService;
+
+        public AuthService(
+            DataContext context,
+            JwtFactory jwtFactory,
+            IMapper mapper,
+            IValidator<User> validator,
+            ConfirmationTokenService tokenService)
+        {
+            _context = context;
+            _mapper = mapper;
+            _jwtFactory = jwtFactory;
+            _validator = validator;
+            _tokenService = tokenService;
+        }
+
+        public async Task<UserDto> Login(UserLoginDto loginInfo)
+        {
+            var userEntity = await _context.Users.FirstOrDefaultAsync(x => x.Email == loginInfo.Email)
+                ?? throw new ValidationException("No user with given email");
+
+            if (!userEntity.IsEmailConfirmed)
+            {
+                if (!_context.ConfirmationTokens.Any(x => x.UserId == userEntity.Id))
+                {
+                    var token = await _tokenService.CreateConfirmationToken(userEntity, TokenKind.EmailConfirmation);
+                    await _tokenService.SendConfirmationEmail(token);
+                }
+                throw new EmailNotConfirmedException(userEntity.Email);
+            }
+
+            if (!SecurityHelper.ValidatePassword(loginInfo.Password, userEntity.Password, userEntity.Salt))
+                throw new ValidationException("Invalid password");
+
+            return _mapper.Map<UserDto>(userEntity);
+        }
+
+        public async Task<UserDto> Login(Guid emailToken)
+        {
+            var confToken = await _tokenService.ConfirmToken(emailToken, TokenKind.EmailConfirmation);
```

Comment:
We do not handle exceptions, thrown by `ConfirmToken` at the moment. Please, fix it and look for other places where similar problem may occur.

Reply:
`ConfirmToken` throws only ValidationExceptions which are caught by `ExceptionMiddleware` and transformed into 400 Bad Req responses

Reply:
Ok then. Don't think that it's the best way for handling, but it's out of scope of this ticket.

Reply:
My thoughts were that we should give users links to frontend endpoints which gonna send request to backend and show appropriate response, if it's Bad Req we could show error message, if everything is okay we'll proceed with the flow